invenio/modules/documents/config.py

```diff
@@ -36,5 +34,5 @@
     'model': 'Document',
     'host': "localhost",
     'port': 27017,
-    'database': config.CFG_DATABASE_NAME,
+    'database': LocalProxy(lambda: cfg['CFG_DATABASE_NAME']),
```

Comment:
Actually this is a bit tricky!
I agree that the correct way is to put it here like: `LocalProxy(lambda: cfg['CFG_DATABASE_NAME'].encode('utf-8'))`, but the problem is with `pymongo`.
Internally when someone tries to create a new table it does:
```python
if not isinstance(name, basestring):
    raise TypeError("name must be an instance "
                              "of %s" % (basestring.__name__,))
```
And this is of type `<class 'werkzeug.local.LocalProxy'>` unless I call encode before.

Probably there is a better to do it.